src/minichain/genesis.py

```
@@ -0,0 +1,81 @@
+@dataclass(frozen=True)
+class GenesisConfig:
+    """Configurable parameters for building genesis artifacts."""
+
+    initial_balances: dict[str, int] = field(default_factory=dict)
+    timestamp: int = 1_739_000_000
+    difficulty_target: int = (1 << 255) - 1
+    version: int = 0
+
+    def validate(self) -> None:
+        if self.timestamp < 0:
+            raise ValueError("Genesis timestamp must be non-negative")
```

Ruff (0.15.2)

[warning] Avoid specifying long messages outside the exception class

(TRY003)